channels/tests/test_management.py

```diff
@@ -0,0 +1,133 @@
+        call_command('runworker', '--layer', 'fake_channel')
+        mocked_worker.assert_called_with(callback=None,
+                                         only_channels=None,
+                                         channel_layer=mock.ANY,
+                                         exclude_channels=None)
+
+    @mock.patch('channels.management.commands.runworker.Worker')
+    def test_runworker_verbose(self, mocked_worker):
+        call_command('runworker', '--layer',
+                     'fake_channel', '--verbosity', '2')
+        # Increasing verbosity adds the logger callback
+        mocked_worker.assert_called_with(callback=mock.ANY,
+                                         only_channels=None,
+                                         channel_layer=mock.ANY,
+                                         exclude_channels=None)
+
+
+class RunServerTests(TestCase):
+
+    def setUp(self):
+        import channels.log
+        self.stream = StringIO()
+        channels.log.handler = logging.StreamHandler(self.stream)
+
+    @mock.patch('sys.stderr', new_callable=StringIO)
+    @mock.patch('sys.stdout', new_callable=StringIO)
+    @mock.patch('channels.management.commands.runserver.Server')
+    @mock.patch('channels.management.commands.runworker.Worker')
+    def test_runserver_basic(self, mocked_worker, mocked_server, mock_stdout, mock_stderr):
+        call_command('runserver', '--noreload')
+        mocked_server.assert_called_with(port=8000, signal_handlers=True, http_timeout=60,
+                                         host='127.0.0.1', action_logger=mock.ANY, channel_layer=mock.ANY)
+
+    @mock.patch('sys.stderr', new_callable=StringIO)
+    @mock.patch('sys.stdout', new_callable=StringIO)
+    @mock.patch('channels.management.commands.runserver.Server')
+    @mock.patch('channels.management.commands.runworker.Worker')
+    def test_runserver_nostatic(self, mocked_worker, mocked_server, mock_stdout, mock_stderr):
+        with self.settings(DEBUG=True, STATIC_URL='/static/'):
+            call_command('runserver', '--noreload')
+            mocked_server.assert_called_with(port=8000, signal_handlers=True, http_timeout=60,
+                                             host='127.0.0.1', action_logger=mock.ANY, channel_layer=mock.ANY)
+
+            call_command('runserver', '--noreload', 'localhost:8001')
+            mocked_server.assert_called_with(port=8001, signal_handlers=True, http_timeout=60,
+                                             host='localhost', action_logger=mock.ANY, channel_layer=mock.ANY)
+
+    @mock.patch('sys.stderr', new_callable=StringIO)
+    @mock.patch('sys.stdout', new_callable=StringIO)
+    @mock.patch('channels.management.commands.runserver.Server')
+    def test_runserver_noworker(self, mocked_server, mock_stdout, mock_stderr):
+        '''
+        No need to Mock the Worker because it should not be used.
+        '''
+        call_command('runserver', '--noreload', '--noworker')
```

Comment:
What is this testing? You seem to have mocked out the command being tested so it'll vacuously be true.